data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,498 @@
+                        clean_consts.DESTINATION_DATASET: sandbox_dataset,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        'type': 'sandbox'
+                    })
+                    clean_query = CLEAN_QUERY_END_DATE.render(
+                        project=date_row.project_id,
+                        dataset=date_row.dataset_id,
+                        table=date_row.table,
+                        pid=pid,
+                        deactivated_pids_project=pids_project_id,
+                        deactivated_pids_dataset=pids_dataset_id,
+                        deactivated_pids_table=pids_table,
+                        end_date_column=date_row.end_date_column,
+                        start_date_column=date_row.start_date_column)
+                    queries_list.append({
+                        clean_consts.QUERY: clean_query,
+                        clean_consts.DESTINATION_DATASET: date_row.dataset_id,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        clean_consts.DISPOSITION: bq_consts.WRITE_TRUNCATE,
+                        'type': 'retraction'
+                    })
+            else:
+                # break out of loop to create query, if pid does not exist in table
+                continue
+    LOGGER.info(
+        f"Query list complete, retracting ehr deactivated PIDS from the following datasets: "
+        f"{dataset_list}")
+    return queries_list
+
+
+def run_queries(queries, client):
+    """
+    Function that will perform the retraction.
+
+    :param queries: list of queries to run retraction
+    """
+    query_job_ids = []
+    for query_dict in queries:
+        if query_dict['type'] == 'sandbox':
+            LOGGER.info(
+                f"Writing rows to be retracted to, using query {query_dict['query']}"
+            )
+            response = client.query(query_dict['query'])
+            LOGGER.info(
+                f"{query_dict['destination_table_id']} table written to {query_dict['destination_dataset_id']}"
+            )
+            query_job_id = response.job_id
+            query_job_ids.append(query_job_id)
+        else:
+            LOGGER.info(
+                f"Truncating table with clean data, using query {query_dict['query']}"
+            )
+            response = client.query(query_dict['query'])
+            LOGGER.info(
+                f"{query_dict['destination_table_id']} table updated with clean rows in "
+                f"{query_dict['destination_dataset_id']}")
+            query_job_id = response.job_id
+            query_job_ids.append(query_job_id)
+
+    incomplete_jobs = response.exception()
+    if incomplete_jobs is not None:
+        count = len(incomplete_jobs)
+        if incomplete_jobs:
+            LOGGER.info(f"Failed on {count} job ids {incomplete_jobs}")
+            LOGGER.info("Terminating retraction")
+            raise incomplete_jobs
```

Comment:
your responses are generated in your for loop and you're not saving them to an outside list or variable.  perhaps you want a list to append response objects to.  Then you would loop through the list of response objects to make sure they are finished and finished without exceptions.  You could check with:
```
# assuming response_list is a list of response objects
for resp in response_list:
    if resp.result():
        if resp.exception():
            # logging statements or exception aggregation or your desired custom code
        else:
            # job finished without errors
    else:
        # job isn't finished yet
```

Reply:
I've added `job_result()` after each `client_query()` so it waits for the job to finish. If there is an exception it will append to the `incomplete_jobs` list, if not, it will then hit the `LOGGER.info` stating which table was written with the corresponding query. Please take a look and let me know if you see any adjustments that need to be made.